### PR TITLE
Gui hooks for entities and tileentities

### PR DESCRIPTION
**Pull request by Nightenom**
# Changes proposed in this pull request:
- adds ability to render gui above entities and tileentities in the same way as vanilla entity name is rendered
- gui scale is 100 px = 1 block
- naming might be weird since my brain is having a weird week, feel free to suggest changes

Review please

### CODE REVIEW

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    @Override
+    public void init()
+    {
+        // noop
+    }
+
+    @Override
+    public void tick()
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:64]: does this have to tick?

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+import com.ldtteam.blockout.BOScreen;
+import com.ldtteam.blockout.hooks.TriggerMechanism.Type;
+import com.ldtteam.blockout.views.ScrollingList;
+import com.mojang.blaze3d.matrix.MatrixStack;
+
+/**
+ * Screen wrapper.
+ */
+public class HookScreen extends BOScreen
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:11]: this class is missing @Overrides

File: src/main/java/com/ldtteam/blockout/hooks/HookManager.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+import java.util.Set;
+import java.util.function.Function;
+import com.ldtteam.blockout.hooks.TriggerMechanism.Type;
+import com.ldtteam.structurize.api.util.constant.Constants;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.client.Minecraft;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+public abstract class HookManager<T, U extends IForgeRegistryEntry<U>, K>
```
Comment (Raycoms) [inline src/main/java/com/ldtteam/blockout/hooks/HookManager.java:26]: this should also have an interface with javadoc for the methods

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:11]: nah, those methods are new methods, if you mean some concrete methods then name them (I think I have covered all of them)

File: src/main/java/com/ldtteam/blockout/hooks/HookManager.java (hunk 3 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/hooks/HookManager.java:26]: you mean something like IHookManager? why? this is not part of visible API imho (basically, you only want to extend it)

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java (hunk 1 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:64]: the gui can have one scrolling list, so probably yes

File: src/main/java/com/ldtteam/blockout/hooks/HookManager.java (hunk 3 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/blockout/hooks/HookManager.java:26]: Mostly to have a nice place for all the docu

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java (hunk 2 above)
Comment (Raycoms) [reply src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:11]: ah okay, those are missing some javadoc then =D, I thought the render method at least

File: src/main/java/com/ldtteam/blockout/hooks/HookManager.java (hunk 3 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/hooks/HookManager.java:26]: Like there is nothing to document, can document registerInternal but that's all, tick and render are quite obvious

File: src/main/java/com/ldtteam/blockout/hooks/HookScreen.java (hunk 2 above)
Comment (Nightenom) [reply src/main/java/com/ldtteam/blockout/hooks/HookScreen.java:11]: kk

File: src/main/java/com/ldtteam/blockout/Loader.java [hunk 4]
```diff
@@ -180,8 +182,7 @@ private static void createFromXML(final InputSource input, final View parent)
     {
         try
         {
-            final DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
-            final DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            final DocumentBuilder dBuilder = DOCUMENT_BUILDER_FACTORY.newDocumentBuilder();
```
Comment (Nightenom) [inline src/main/java/com/ldtteam/blockout/Loader.java:185]: This should be reusable too, but I'm not sure so let's rather be safe

**Review by marchermans — APPROVED** [review]

**Pull request merged**